src/components/fields/money-field/money-field.js

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import has from 'lodash.has';
+import requiredIf from 'react-required-if';
+import Constraints from '../../constraints';
+import Spacings from '../../spacings';
+import FieldLabel from '../../field-label';
+import MoneyInput from '../../inputs/money-input';
+import createSequentialId from '../../../utils/create-sequential-id';
+import FieldErrors from '../../field-errors';
+import filterDataAttributes from '../../../utils/filter-data-attributes';
+
+const sequentialId = createSequentialId('text-field-');
+
+const hasErrors = errors => errors && Object.values(errors).some(Boolean);
+
+class MoneyField extends React.Component {
+  static displayName = 'MoneyField';
+
+  static getAmountInputId = MoneyInput.getAmountInputId;
+  static getCurrencyDropdownId = MoneyInput.getCurrencyDropdownId;
+  static convertToMoneyValue = MoneyInput.convertToMoneyValue;
+  static parseMoneyValue = MoneyInput.parseMoneyValue;
+  static isEmpty = MoneyInput.isEmpty;
+  static isHighPrecision = MoneyInput.isHighPrecision;
+  static isTouched = MoneyInput.isTouched;
+
+  static propTypes = {
+    // MoneyField
+    id: PropTypes.string,
+    horizontalConstraint: PropTypes.oneOf(['xs', 's', 'm', 'l', 'xl', 'scale']),
+    errors: PropTypes.shape({
+      missing: PropTypes.bool,
+    }),
+    renderError: PropTypes.func,
+    isRequired: PropTypes.bool,
+    touched: PropTypes.shape({
```

Comment:
Our other fields accept `isTouched` instead. I did not want to accept `isTouched` here as it would be harder on the consumer.

The touched object looks like this: `touched: { somePrice: { amount: true, currencyCode: true } }`.

We want consumers to be able to do:

```jsx
<MoneyField name="somePrice" touched={formik.touched.somePrice} />
```

If we wanted to accept an `isTouched` prop we'd have to do:

```jsx
<MoneyField
  name="somePrice"
  isTouched={MoneyField.isTouched(formik.touched.somePrice)}
/>
```

which seems awkward.

---

I am thinking of changing `isTouched` in `TextField` and `NumberField` to be just `touched` as well.

Reply:
> I am thinking of changing isTouched in TextField and NumberField to be just touched as well.

I wouldn't have minded the isTouched to be honest. Just suggesting to do the backflip the other way round. We introduce isTouched for now, keeping all fields consistent, to then refactor it to `touched` with all fields in one go?

Reply:
Maybe there is a misunderstanding. If we use wanted to use `isTouched`, then every consumer would have had to do:

```js
<MoneyField
  isTouched={MoneyField.isTouched(formik.touched.price)}
/>
```

Whereas now they can do

```js
<MoneyField touched={formik.touched.price} />
```

Accepting the whole `touched` object further keeps the possibility to use the information in the `touched` state. For example, when we receive a `missing` error we can inspect the `value` and the `touched` state to display an error accordingly - something we could not do properly when we only accept a boolean `isTouched` as we'd lose the information of whether `amount` or `currencyCode` was touched.

Reply:
> > I am thinking of changing isTouched in TextField and NumberField to be just touched as well.
>
> I wouldn't have minded the isTouched to be honest. Just suggesting to do the backflip the other way round. We introduce isTouched for now, keeping all fields consistent, to then refactor it to `touched` with all fields in one go?

I don't fully get the question. `TextField` and `NumberField` have one input only. `MoneyField` uses two "inputs" (the currency dropdown and the number input).

`TextField` and `NumberField` have one touched state only, while `MoneyField` has two (one for currency, one for amount).

I was thinking whether it would make sense to keep the name consistent and always accept a `touched` prop on fields. It would accept a boolean for `TextField` and `NumberField`, but it would accept an object (`{ amount, currencyCode }`) for `MoneyField`.

For now it is like this:

```js
<TextField isTouched={true} />
<NumberField isTouched={true} />
<MoneyField touched={{ amount: true, currencyCode: false }} />
```

Eventually, it could become

```js
<TextField touched={true} />
<NumberField touched={true} />
<MoneyField touched={{ amount: true, currencyCode: false }} />
```

Reply:
Okay okay. `touched` is fine.

Reply:
Cool :) I didn't get your question so I just tried to give the answer to every possible question :P

Using `touched` from the beginning avoids the refactoring on the consumer and allows us to do nicer UX inside `MoneyField` right away. There are only two fields with `isTouched` so far and both are not used yet, so refactoring the fields is the lesser evil :) 

I wrote the comment initially to find out whether we're okay with always using `touched` even if it contains a boolean, as this is kind of against our current prop naming conventions. Sorry that this wasn't more clear. If we're okay with it, I'd rename `isTouched` in `NumberField` and `TextField` now.

Reply:
Alright, then. Fine here and lets refactor the other two then before used :)